components/datadog/agent/kubernetes.go

```diff
@@ -0,0 +1,55 @@
+package agent
+
+import (
+	"github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/DataDog/test-infra-definitions/components/datadog/kubernetesagentparams"
+)
+
+type KubernetesAgentOutput struct {
+	components.JSONImporter
+
+	AgentInstallName        string `json:"agentInstallName"`
+	AgentInstallNameWindows string `json:"agentInstallNameWindows"`
+}
+
+// DockerAgent is a Docker installer on a remote Host
+type KubernetesAgent struct {
+	pulumi.ResourceState
+	components.Component
+
+	AgentInstallName        pulumi.StringOutput `pulumi:"agentInstallName"`
+	AgentInstallNameWindows pulumi.StringOutput `pulumi:"agentInstallNameWindows"`
+}
+
+func (h *KubernetesAgent) Export(ctx *pulumi.Context, out *KubernetesAgentOutput) error {
+	return components.Export(ctx, h, out)
+}
+
+func NewKubernetesAgent(e config.CommonEnvironment, clusterName string, kubeProvider *kubernetes.Provider, options ...kubernetesagentparams.Option) (*KubernetesAgent, error) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Could this be used in the eks run function at https://github.com/DataDog/test-infra-definitions/blob/nplanel/fix-docker-publish/scenarios/aws/eks/run.go#L217-L236 ?

Reply:
I think it _could_ be used in this run function. However I think it is easier to use it with a Provider (for [example](https://github.com/DataDog/datadog-agent/blob/kfairise/kindvm-provisioner/test/new-e2e/pkg/environments/aws/kubernetes/kindvm.go#L113)) that will automatically register the output. So you do not need to add explicit export to the Pulumi stack like that:
```
ctx.Export("agent-linux-helm-install-name", helmComponent.LinuxHelmReleaseName)
ctx.Export("agent-linux-helm-install-status", helmComponent.LinuxHelmReleaseStatus)
```